bootstrap.d/games-fps.yml

```diff
@@ -0,0 +1,110 @@
+sources:
+  - name: dsda-doom
+    subdir: ports
+    git: 'https://github.com/kraflab/dsda-doom.git'
+    tag: 'v0.22.1'
+    version: '0.22.1'
+
+tools:
+  - name: host-rdatawad
+    from_source: dsda-doom
+    tools_required:
+      - host-cmake
+    configure:
+      - args: ['echo', '@THIS_SOURCE_DIR@']
+      - args:
+        - 'cmake'
+        - '-GNinja'
+        - '-DCMAKE_BUILD_TYPE=Release'
+        - '-DBUILD_RDATAWAD_ONLY=ON'
+        - '-DBUILD_GL=OFF'
+        - '@THIS_SOURCE_DIR@/prboom2'
+    compile:
+      - args: ['ninja', 'rdatawad']
+    install:
+      # We perform the install manually (and also patch the generated CMake file)
+      # since you cannot install rdatawad on it's own, and we don't want to build
+      # the whole dsda-doom for the host.
+      - args: ['mkdir', '-p', '@PREFIX@/']
+      - args: ['cp', '@THIS_BUILD_DIR@/data/rdatawad', '@PREFIX@/rdatawad']
+      - args: |
+          sed -e 's|@THIS_BUILD_DIR@/data|@PREFIX@|' @THIS_BUILD_DIR@/ImportExecutables.cmake \
+              > @PREFIX@/ImportExecutables.cmake
+
+packages:
+  - name: dsda-doom
+    architecture: '@OPTION:arch@'
+    metadata:
+      summary: Modern DOOM source port with extra tooling for demos and speedrunning
+      description: Modern DOOM source port based on PrBoom+ with support for new standards and extra tooling for demo playback, demo recording and speedrunning.
+      spdx: 'GPL-2.0-or-later'
+      website: 'https://github.com/kraflab/dsda-doom'
+      maintainer: 'Kacper Słomiński <qookie@managarm.org>'
+      categories: ['games-fps']
+    from_source: dsda-doom
+    tools_required:
+      - host-cmake
+      - host-pkg-config
+      - system-gcc
+      - host-rdatawad
+      - virtual: pkgconfig-for-target
+        triple: "@OPTION:arch-triple@"
+    pkgs_required:
+      - mlibc
+      - zlib
+      - pcre
+      - sdl2
+      - glu
+    configure:
+      - args:
+        - 'cmake'
+        - '-GNinja'
+        - '-DCMAKE_TOOLCHAIN_FILE=@SOURCE_ROOT@/scripts/CMakeToolchain-@OPTION:arch-triple@.txt'
+        - '-DCMAKE_INSTALL_PREFIX=/usr'
+        - '-DCMAKE_BUILD_TYPE=Release'
+        - '-DIMPORT_EXECUTABLES=@BUILD_ROOT@/tools/host-rdatawad/ImportExecutables.cmake'
+        - '@THIS_SOURCE_DIR@/prboom2'
+    build:
+      - args: ['ninja']
+      - args: ['ninja', 'install']
+        environ:
+          DESTDIR: '@THIS_COLLECT_DIR@'
+        quiet: true
+
+  - name: tyr-quake
+    architecture: '@OPTION:arch@'
+    source:
+      subdir: 'ports'
+      git: 'git://disenchant.net/tyrquake'
```

Comment:
Don’t we want to use https here too?

Reply:
This is the link given on the website, and cloning from `https://disenchant.net/git/tyrquake.git` (which uses the dumb http transport) is significantly slower than cloning from `git://disenchant.net/tyrquake` (2min vs 7s)

Reply:
doesn't matter, no need to use a nontls protocol

Reply:
Hm, good point

Reply:
... or https://disenchant.net/files/engine/tyrquake-0.69.tar.gz. Though, we lack c-sum checking for now. I'll put that on my xbstrap todo list (or on the issue tracker).

Reply:
Yeah after reading this convo I think we want to use https here too, especially since mirroring is a thing. If you change to that I can approve and merge this PR.

Reply:
What's the advantage of https over git? Why should we switch even if qookie tried it out and it's much slower?

Reply:
Is the git protocol unencrypted by default? I thought git meant "git over SSH" but maybe I'm wrong here.

Reply:
Looks like it's unencrypted:
```
connect(3, {sa_family=AF_INET, sin_port=htons(9418), sin_addr=inet_addr("112.213.32.136")}, 16) = 0
setsockopt(3, SOL_SOCKET, SO_KEEPALIVE, [1], 4) = 0
dup(3)                                  = 4
write(4, "003d", 4)                     = 4
write(4, "git-upload-pack /tyrquake\0host=d"..., 57) = 57
```

Reply:
Nope, `git://` is git's smart protocol over plain TCP. Here's an example packet:

```
0000   30 30 33 64 67 69 74 2d 75 70 6c 6f 61 64 2d 70   003dgit-upload-p
0010   61 63 6b 20 2f 74 79 72 71 75 61 6b 65 00 68 6f   ack /tyrquake.ho
0020   73 74 3d 64 69 73 65 6e 63 68 61 6e 74 2e 6e 65   st=disenchant.ne
0030   74 00 00 76 65 72 73 69 6f 6e 3d 32 00            t..version=2.
```

I usually see a slowdown of about 5-10% max (which is probably attributable to TLS overhead) but in this case it's over 15x slower. I assume the `https://` server in this case isn't a `git-http-backend` or such.

Reply:
One slight problem, xbstrap can't clone over the dumb transport since it needs shallow clones:
```
xbstrap: fetch tyr-quake [1/8]
fatal: dumb http transport does not support shallow capabilities
xbstrap: Action fetch of source tyr-quake failed
```

Reply:
There is a `no_shallow_fetch: true` property to work around this issue.